gradle/utility.gradle

```diff
@@ -0,0 +1,35 @@
+project.ext.ideaActive = System.getProperty('idea.active') == 'true'
+
+kotlin {
+    targets {
+        def manager = project.ext.hostManager
+        def linuxEnabled = manager.isEnabled(presets.linuxX64.konanTarget)
+        def macosEnabled = manager.isEnabled(presets.macosX64.konanTarget)
+        def winEnabled = manager.isEnabled(presets.mingwX64.konanTarget)
+
+        project.ext.isLinuxHost = linuxEnabled
+        project.ext.isMacosHost = macosEnabled
+        project.ext.isWinHost = winEnabled
+
+        if (project.ext.ideaActive) {
+            def ideaPreset = presets.linuxX64
+            if (macosEnabled) ideaPreset = presets.macosX64
+            if (winEnabled) ideaPreset = presets.mingwX64
+
+            project.ext.ideaPreset = ideaPreset
+        }
+    }
+}
+
+def java_version = System.properties["java.version"]
```

Comment:
as this is used in multiple projects - perhaps we should source this from some build/meta project do keep DRY